proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/saml/HubResponseTranslator.java

```diff
@@ -43,9 +45,16 @@ public HubResponseTranslator(
     Response getTranslatedHubResponse(HubResponseContainer hubResponseContainer, CountryMetadataResponse countryMetadataResponse) {
         final List<EidasAttributeBuilder> eidasAttributeBuilders = new ArrayList<>();
 
-        final String pid = hubResponseContainer.getPid()
-                .map(p -> String.format(PID_PREFIX, countryMetadataResponse.getCountryCode(), p))
-                .orElse(null);
+        final String pid;
+        final SecureRandomIdentifierGenerationStrategy idGeneratorStrategy = new SecureRandomIdentifierGenerationStrategy();
```

Comment:
make this a `private static final` field so it only gets instantiated once. No need to do it in constructor for testing (@idrop )